react-validated-input-component/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,42 @@
+import { ChangeEvent, useState } from 'react';
+import { FcCheckmark, FcCancel } from 'react-icons/fc';
+import { ErrorMessage } from './ErrorMessage';
+
+function isValid(pw: string): boolean {
```

Comment:
It would be nice to figure out a way to determine if the password is valid without using both a regex here and a bunch of tests in ErrorMessage.

I would imagine that the best way to do that is to have several top-level functions that both components can import.